### PR TITLE
trying to implement admin functionality -> it fails for now

### PR DESCRIPTION
**Pull request by adba3894**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,19 +21,34 @@ apply plugin: "io.spring.dependency-management"
 group = "com.example"
 version = "0.1.0"
 
+//dependencies {
```
Comment (PovilasKM) [inline build.gradle]: why add commented lines?

File: src/main/java/com/good/works/summer/project/Application.java [hunk 2]
```diff
@@ -38,6 +39,10 @@ CommandLineRunner citiesRunner(CityService cityService) {
 		};
 	}
 
+	@Bean
+	public BCryptPasswordEncoder bCryptPasswordEncoder() {
```
Comment (PovilasKM) [inline src/main/java/com/good/works/summer/project/Application.java]: Is this needed here?

File: src/main/java/com/good/works/summer/project/controller/HelloAdminController.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.good.works.summer.project.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HelloAdminController {
```
Comment (PovilasKM) [inline src/main/java/com/good/works/summer/project/controller/HelloAdminController.java]: Is this controller needed?

File: src/main/java/com/good/works/summer/project/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    public UserController(ApplicationUserRepository applicationUserRepository,
+                          BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.applicationUserRepository = applicationUserRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+
+    @PostMapping("/sign-up")
+    public void signUp(@RequestBody ApplicationUser user) {
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
```
Comment (PovilasKM) [inline src/main/java/com/good/works/summer/project/controller/UserController.java]: Move the logic to a service

File: src/main/java/com/good/works/summer/project/entities/ApplicationUser.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.good.works.summer.project.entities;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class ApplicationUser {
```
Comment (PovilasKM) [inline src/main/java/com/good/works/summer/project/entities/ApplicationUser.java]: Why applicationUser and not just user?

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -6,6 +6,7 @@ spring.datasource.url=jdbc:postgresql://ec2-54-75-230-41.eu-west-1.compute.amazo
 spring.datasource.username=rudmnperrltvai
 spring.datasource.password=511e64313bf5583e0064f9511b92321d0ef4423d4d817228dc8856135600995d
 spring.datasource.driver-class-name=org.postgresql.Driver
+#spring.jpa.hibernate.ddl-auto=update
```
Comment (PovilasKM) [inline src/main/resources/application.properties]: Why leave commented lines?

**Review by PovilasKM — CHANGES_REQUESTED** [review]
Remove out folder and check if all java files end in empty line.

File: src/main/java/com/good/works/summer/project/constants/SecurityConstants.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.good.works.summer.project.constants;
+
+public class SecurityConstants {
+    public static final String SECRET = "SecretKeyToGenJWTs";
+    public static final long EXPIRATION_TIME = 864_000_000; // 10 days
+    public static final String TOKEN_PREFIX = "Bearer ";
+    public static final String HEADER_STRING = "Authorization";
+    public static final String SIGN_UP_URL = "/users/sign-up";
```
Comment (IndirasM) [inline src/main/java/com/good/works/summer/project/constants/SecurityConstants.java]: Any reason why the url includes /users and signup includes a dash?

**Review by IndirasM — CHANGES_REQUESTED** [review]

**Review by PovilasKM — APPROVED** [review]
But next PR everything will have to be fixed.

**Pull request merged**